app/grants/utils.py

```diff
@@ -392,88 +395,73 @@ def toggle_user_sybil(sybil_users, non_sybil_users):
-        ml_non_sybil = ml_non_sybil[['handle', 'flag_type_x', 'notes']]
-        ml_non_sybil = ml_non_sybil.rename({'handle': 'handle', 'flag_type_x': 'label', 'notes': 'comment'}, axis = 1, inplace = True)
-        non_sybil_df = non_sybil_df.append(ml_non_sybil)
-
-        '''
-        conversion of all the data so that it can be pushed to the toggle_user_sybil endpoint
-        '''
-        #sybil_df = ml_df[ml_df['ml_score'] >= 0.9 and ml_df['flag_type'] != 'Human']
-        sybil_users = sybil_df.to_dict('records')
-        non_sybil_users = non_sybil_df.to_dict('records')
-
-        # print('=================SYBIL=================')
-        # print(sybil_users)
-        # print('=================NON SYBIL=================')
-        # print(non_sybil_users)
-
-        toggle_user_sybil(sybil_users, non_sybil_users)
+def bsci_script(csv: str) -> Union[FlaggingScriptOutput, None]:
+    """
+    Generate records of sybil / non-sybil users based
+    on the CSV output as provided by BSci detection pipeline.
+    """
 
+    # Assumptions
+    RENAME_MAP = {'notes': 'comment'}
+    ML_THRESHOLD = 0.8
+    EVAL_THRESHOLD = 0.8
+    HEURISTIC_THRESHOLD = 0.5
+    
+    # Read CSV
+
+    try: 
+        df = (pd.read_csv(csv)
+                .assign(is_sybil=None)
+                .assign(label=None)
+                .rename(columns=RENAME_MAP))
+        
+        # Get label domains
+        rows_with_evaluation = ~pd.isnull(df.evaluation_score) 
+        rows_with_heuristic = ~pd.isnull(df.heuristic_score)
+        rows_with_prediction = ~pd.isnull(df.prediction_score)
+        
+        labels_by_evaluation = rows_with_evaluation
+        labels_by_heuristic = (rows_with_heuristic & (rows_with_heuristic 
+                                                    ^ labels_by_evaluation))
+        labels_by_prediction = (rows_with_prediction & (rows_with_prediction 
+                                                        ^ (labels_by_heuristic | 
+                                                        labels_by_evaluation)))
+        
+        # Assign final `is_sybil` markings according to a priorization criteria
+        df.loc[labels_by_evaluation, 'is_sybil'] = df[labels_by_evaluation].evaluation_score
+        df.loc[labels_by_evaluation, 'label'] = "Human Evaluation"
+        
+        df.loc[labels_by_heuristic, 'is_sybil'] = df[labels_by_heuristic].heuristic_score
+        df.loc[labels_by_heuristic, 'label'] = "Heuristics"
+        
+        df.loc[labels_by_prediction, 'is_sybil'] = df[labels_by_prediction].prediction_score
+        df.loc[labels_by_prediction, 'label'] = "ML Prediction"
+        
+        # Generate dict records
+        sybil_records = df.query('is_sybil == True').to_dict('records')
+        non_sybil_records = df.query('is_sybil == False').to_dict('records')
+        
+        sybil_records = [ToggleUser(**d) for d in sybil_records]
+        non_sybil_records = [ToggleUser(**d) for d in non_sybil_records]
+        
```

Comment:
Could we add back the call to `toggle_user_sybil(sybil_users, non_sybil_users)`:

```suggestion
                toggle_user_sybil(sybil_records, non_sybil_records)
                
```

Reply:
Or alter [app/grants/tasks.py -> process_bsci_sybil_csv()](https://github.com/gitcoinco/web/blob/5db13f06b7b3dedaa683f504f045006d7fe95b8a/app/grants/tasks.py#L443) to make this call with the return data.